src/index.js

```diff
@@ -387,8 +369,17 @@ Penpal.connectToParent = ({ parentOrigin, methods = {} }) => {
       child.removeEventListener(MESSAGE, handleMessageEvent);
       reject('Child: Connection destroyed');
     })
-  });
 
+    const sendHandshakeMessage = function() {
+      parent.postMessage({
+	    penpal: HANDSHAKE,
+	    methodNames: Object.keys(methods),
+      }, document.referrer);
```

Comment:
Do you mind adding a `log('Child: Sending handshake');` right before the handshake message is sent?